language/move-native/src/conv.rs

```diff
@@ -119,14 +118,14 @@ pub struct MoveBorrowedRustVecMut<'mv, T> {
 
 impl<'mv, T> Drop for MoveBorrowedRustVec<'mv, T> {
     fn drop(&mut self) {
-        let rv = mem::replace(&mut self.inner, Vec::new());
+        let rv = mem::take(&mut self.inner);
         mem::forget(rv);
```

Comment:
No. mem::forget takes `rv` by value so it is no longer in scope to be passed to any other function.

We call forget here because `rv` is type `Vec`, which owns its interior buffer pointer, but this `Vec` was temporarily fabricated from a `MoveUntypedVector`, which _actually_ owns that buffer. So forgetting the Vec quietly destroys it without running the Vec destructor - so that the original MoveUntypedVector can continue owning that buffer.

It is never required to call `mem::drop` because it does nothing but run the value's destructor, which would have also happened without passing it to `mem::drop`. Calling `mem::drop` is sometimes done for clarity, and sometimes to force a value to be dropped before it would naturally go out of scope.

Reply:
The only reason to call `mem::forget` is to not run a destructor.